src/Nordea/Components/Checkbox.elm

```diff
@@ -222,11 +223,13 @@ view attrs (Checkbox config) =
             , notDisabledSpecificStyling
             , appearanceStyle
             ]
-            :: attrs
+         ]
+            ++ attrs
         )
         [ Html.input
             [ type_ "checkbox"
             , name config.name
+            , Attrs.id config.name
```

Comment:
An issue with using the name directly as the id is that the name in some cases should _not_ be unique. The name is required to be the same for groups of inputs, like radio buttons and checkboxes, etc., for the browser to know that they are semantically connected.

It can be solved by adding it as a required separate parameter, and provide it manually for every input. But the problem is avoided altogether if we make the input a child of the label instead.

Reply:
As per the spec, it doesn't mention anything about using both nesting and `for` as far as I can tell (but the words `either` and `or`). So not sure if it is valid 🤔

https://www.w3.org/TR/html401/interact/forms.html#h-17.9.1

<img width="761" alt="Screenshot 2022-10-27 at 11 17 54" src="https://user-images.githubusercontent.com/42603101/198245346-66d5a36a-c24b-4951-9a67-9269743cb101.png">

Reply:
Good point! We'll add it for `input`s and `label`s on the same level🙏

> problem is avoided altogether if we make the input a child of the label instead

- Every major component library I find have labels and inputs on the same level in the DOM ([NAV](https://aksel.nav.no/designsystem/komponenter/textarea), [Entur](https://design.entur.org/komponenter/skjemaelementer/textfield), [Vy](https://spor.cloud.vy.no/komponenter/inputfelt))
- Documentation on w3's pages explicitly recommend to [use explicit labelling](https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-implicitly#associating-labels-implicitly)
- Having the `input` and the `label` on the same level in the DOM makes the elm component library so much easier to use for us as programmers.

I can't really see any reasons why we should stick to having the input as a child of the label.

Reply:
> Every major component library I find have labels and inputs on the same level in the DOM ([NAV](https://aksel.nav.no/designsystem/komponenter/textarea), [Entur](https://design.entur.org/komponenter/skjemaelementer/textfield), [Vy](https://spor.cloud.vy.no/komponenter/inputfelt))

The Vy label is not connected to the input, thats a bug 😛

> Having the input and the label on the same level in the DOM makes the elm component library so much easier to use for us as programmers.

The only difference I see is how the DOM ends up:

```html
<div>
    <label for="the-input">the label</label>
    <input id="the-input" />
</div>
```

vs:
```html
<label>
    the label
    <input />
</label>
```
The Elm interface can be identical (altough the first requires an extra param, the id), and both are adhering to the spec. The first one also requires a wrapping div in Elm (unless we make our components return lists instead of single tags), and the id handling adds quite a lot to complexity.



I see your points. My concern is only about the added complexity of the id handling. Is that added complexity worth it when both are adhering to the spec anyway? If it were React, I would be ok with both as the id handling can be totally hidden.

Reply:
Valid point with the complexity. On the other hand I think the components themselves will become less complex as we won't have to have a `Label` component needing to take into account what it contains. This will become much less complex I think.
I also feel a need to follow the recommendations made by [WAI](https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-implicitly#associating-labels-implicitly:~:text=but%20the%20labels.-,Associating%20labels%20implicitly,-In%20some%20situations).
Semantically it is also undoubtedly more correct with them on the same level as the input is not really "part of" the label.

Reply:
I don't agree, but thats ok 🙂

I'm curious about your perspective on the added complexity of the nesting. The way I see it, there is a reduction of complexity with the nested variant. The div is just replaced with a label, we still need to take into account what it contains.

For/id: 
```elm
view props =
    Html.div
        [ css [ someLayoutCss ]]
        [ Html.label [ for props.inputId ] [ Text.bodyTextHeavy |> Text.view [] [ Html.text "Some label" ] ]
        , Html.input [ id props.inputId ] []
        ]
```

Nesting:
```elm
view props =
    Html.label
        [ css [ someLayoutCss ]]
        [ Text.bodyTextHeavy |> Text.view [] [ Html.text "Some label" ]
        , Html.input [] []
        ]
```

This is about whether we want to pay the price of id handling, or be pragmatic. The recommendation is also not about semantics, but screen reader support. It may be a valid point, but I dont know. What if it is based on older outdated screen readers that dont follow the spec; still worth it?

Reply:
We still need to take into account what it contains, but the `Label` component won't be the one having to support everything that can be contained.
I also think it is a problem that we have various components inside our `Label` component and not just the `Html.input`. Which makes the html itself harder to understand. How does a screenreader know which one of the texts inside the `Html.label` element is the actual label?
```elm
Html.label []
    [ Html.div [] [ Html.span [] [ "label" ], Html.span [] [ "requiredness" ]]
    , Html.input [] []
    , Html.div [] [ Html.span [] [ "error" ], Html.span [] [ "counter" ]
    ]
```

> This is about whether we want to pay the price of id handling, or be pragmatic

The price of id handling – is one extra property where needed which I don't think is a cost weighed up against all the benefits (IMO) of doing it this way. Either if we choose to enforce it as mandatory or with a function `withId : String -> Component -> Component`.

Other:
Some guy writes about (grain of salt) why nesting `input` inside `label` isn't supported for older screenreaders. Seems like the solution is still to connect them with `for`, `id` attributes. ([Link](https://dev.to/thejaredwilcurt/the-truth-behind-implicitexplicit-form-labels-2e10))

Reply:
We can take it up in technical town hall and see if anyone has any input. We'll go for the option which most think is the reasonable one😊 (We don't need to make a decision in the meeting itself)